### PR TITLE
improvement: add minimum and maximum transfer amount

### PR DESCRIPTION
**Pull request by 0xstepit**
## Description

To reduce possible spam attack, and improve user experience, this PR adds the following changes:

- Add a minimum transfer amount. The value is defined as a const equal to 1 cent. This const is set specifically for  "uusdc".
- Add a check on the maximum transfer amount equal to the `PerMessageLimitBurn` of the CCTP module.

## Summary by CodeRabbit

- **Documentation**
	- Updated documentation to clearly reflect extended module capabilities.
- **New Features**
	- Introduced enhanced validations for transfers by enforcing updated minimum and maximum amount requirements.
	- Added new methods for handling CCTP operations and querying burn limits.
	- Implemented a dynamic minimum transfer amount in account registration and balance handling.
- **Refactor**
	- Refined internal service naming and error handling to improve overall clarity and reliability.
- **Tests**
	- Expanded test scenarios to cover edge cases and ensure robust transfer validation.

### CODE REVIEW

File: testutil/mocks/autocctp.go [hunk 1]
```diff
@@ -64,8 +64,9 @@ func AutoCCTPKeeper(t testing.TB) (*Mocks, *keeper.Keeper, sdk.Context) {
 		Balances: make(map[string]sdk.Coins),
 	}
 	cctps := CCTPServer{
-		MockCounter: &MockCounter{},
-		Failing:     false,
+		Failing:           false,
+		MaxTransferAmount: 1_000_000_000,
```
Comment (johnletey) [inline testutil/mocks/autocctp.go:68]: Should we make this 1M like on [mainnet](https://api.noble.xyz/circle/cctp/v1/per_message_burn_limits/uusdc)?

**Review by johnletey — APPROVED** [review]
Looking great Ste! One small comment 😄

File: testutil/mocks/autocctp.go (hunk 1 above)
Comment (0xstepit) [reply testutil/mocks/autocctp.go:68]: This is just testing so I believe any value is fine, but If you prefer to have the same value used in mainnet we can update it!

Comment (johnletey) [reply testutil/mocks/autocctp.go:68]: Nope, let's leave it as is 😄
